backend/translation/views.py

```diff
@@ -1,199 +1,246 @@
+            transcript=transcript,
+            target_language=target_language,
+            user=user,
+            payload=payload,
         )
-        # If no translation exists for this user, check if the latest translation can be fetched
-        if queryset is None:
-            if get_latest:
-                queryset = (
-                    Translation.objects.filter(
-                        transcript_id=transcript_id, target_lang=target_lang
-                    )
-                    .order_by("-updated_at")
-                    .first()
-                )
-            else:
-                queryset = None
-
-        # If queryset is empty, return appropriate error
-        if not queryset:
-            return Response(
-                {
-                    "error": "No translation found for the given transcript_id and target_lang."
-                },
-                status=status.HTTP_404_NOT_FOUND,
-            )
 
-        # Serialize and return the data
-        serializer = TranslationSerializer(queryset)
-        return Response(serializer.data)
-
-    @swagger_auto_schema(
-        method="post",
-        request_body=openapi.Schema(
-            type=openapi.TYPE_OBJECT,
-            required=["translation_id", "target_lang", "captions"],
-            properties={
-                "translation_id": openapi.Schema(
-                    type=openapi.TYPE_STRING,
-                    description="An integer identifying the translation instance",
-                ),
-                "target_lang": openapi.Schema(
-                    type=openapi.TYPE_STRING,
-                    description="A string to pass the target language of the translation",
-                ),
-                "captions": openapi.Schema(
-                    type=openapi.TYPE_STRING,
-                    description="A string to pass the translated captions",
-                ),
-            },
-            description="Post request body for projects which have save_type == new_record",
-        ),
-        responses={
-            200: "Translation has been created/updated successfully",
-            400: "Bad request",
-            404: "No translation found for the given transcript_id and target_lang",
-        },
-    )
-    @api_view(["POST"])
-    def post(self, request):
-        # Get the required data from the POST body
-        translation_id = request.data["translation_id"]
-        target_lang = request.data["target_lang"]
-        captions = request.data["captions"]
-        user = request.user
-
-        created = False
-        # Try to get the translation for the given translation_id and target_lang
+    else: 
+        # Try to get the translation for the given translation_id and target_language
         try:
             translation = Translation.objects.get(
-                pk=translation_id, target_lang=target_lang
+                pk=translation_id, target_language=target_language
             )
```

Comment:
Isn't it redundant here to specify the `target_language` ? :)

Reply:
yeah, it's redundant.